feat: better explanation of the Node.js policy (#7652)
withastro/docs: src/content/docs/en/upgrade-astro.mdx

  Once these features are out of their experimental period, they will follow the normal semantic versioning contract.

- **Improvements to the documentation** (e.g. reference and error messages). They are built from source for the `docs` repository. This allows Astro to quickly update docs fixes and improvements in the cases where documentation source content is stored in the main `astro` repository.

### Node.js support and upgrade policies

#### Support

- Astro supports the **latest _extended_ LTS** versions of Node.js.
- Astro can support odd versions of Node.js.

#### Upgrade

> - Odd versions of Node.js can be deprecated and/or dropped when the next even version of Node.js published. This change can occur in a **minor** release of Astro, after an arbitrary period of time decided by the Astro Core team.
> - Upgrade of the minimum **_extended_ LTS** version of Node.js can occur in a **minor** release of Astro.
>   - If a security flaw in Node.js that **affects Astro** is disclosed and fixed, the Core team can bump the minimum version of the **_extended_ LTS** in a **patch** release.
> - Upgrade of minor or major versions of Node.js - **not** extended LTS - occurs only in major versions of Astro.
>   - If a security flaw in Node.js that **affects Astro** is disclosed and fixed, the Core team can bump the minimum version in a **minor** release.

Reviewer: ```suggestion
The following rules define when Astro may deprecate, drop, or add support for versions of Node.js:

- Odd versions of Node.js can be deprecated and/or dropped when the next even version of Node.js published. This change can occur in a **minor** release of Astro, after an arbitrary period of time decided by the Astro Core team.
- Upgrading the minimum **_extended_ LTS** version of Node.js can occur in a **minor** release of Astro.
  - Security exception: If a security flaw in Node.js that **affects Astro** is disclosed and fixed, the Core team can bump the minimum version of the **_extended_ LTS** in a **patch** release.
- Upgrading minor or major versions of Node.js (**not** extended LTS) occurs only in major versions of Astro.
  - Security exception: If a security flaw in Node.js that **affects Astro** is disclosed and fixed, the Core team can bump the minimum version in a **minor** release.
```
I like having an intro sentence to explain what these points are!

### Extended maintenance
 

> The Core team will provide extended maintenance for security fixes, only to the previous major of the current release. This means that of the **current major** is `v4.*`, the Core team will backport security fixes on for `v3.*` versions. 

Reviewer: ```suggestion
The Core team will provide extended maintenance **for security fixes only** for one previous major version. This means that if the **current major** is `v4.*`, the Core team will back port security fixes and issue a new `v3.*` release. 
```
Just checking, but I think last time we issued a new v3.x release, not backported compatibility for everyone on every single previous v3.* version? I would frame it more like above, where "we'll issue an updated v3 version you can update to" and not "you can stay on whichever version of 3 you want"

Author: That's a good point, thank you for the clarification

> The Core team will provide extended maintenance e.g. security fixes, only to the previous major of the current release. This means that of the **current major** is `v4.*`, the Core team will backport security fixes on for `v3.*` versions. 

Reviewer: Just pointing out that this implies that you will provide other maintenance too, and not JUST security fixes. (Not sure I've seen that actually happen? I *think* I've only seen older versions released for security reasons?)

If you really mean "just security fixes" then I would say that! Otherwise, this is pretty vague and makes it sound like v3 is *regularly maintained* too.

Author: That's fair, I'll update it. Only security fixes

  Once these features are out of their experimental period, they will follow the normal semantic versioning contract.

- **Improvements to the documentation** (e.g. reference and error messages). They are built from source for the `docs` repository. This allows Astro to quickly update docs fixes and improvements in the cases where documentation source content is stored in the main `astro` repository.

### Node.js support and upgrade policies

#### Support

> - Astro supports the **latest _extended_ LTS** versions of Node.js.

Reviewer 2: What does "extended LTS" stand for here? On Node's site, they have "Active LTS" and "Maintenance LTS". I suppose this refers to "Maintenance LTS", the longest of the two.
Or is this referring to the extended support that some, like Meteor.js, offer one year beyond the end of the official support from the Node team?

Also, should we link to the Node page so the user can see which version this refers to? I'm guessing that not every user reading this will know from memory which Node version is currently in each support stage.

Author: I will update the terminology, using _Maintainance LTS_

Reviewer 3: I already suggested linking to the node versions diagram page and still think it's a good idea

Reviewer: Can someone drop the link here to the thing you want to link to?

Reviewer 2: https://nodejs.org/en/about/previous-releases#release-schedule

> - Astro supports the **latest _Maintenance_ LTS** versions of Node.js.

Reviewer: ```suggestion
- Astro supports the [**latest _Maintenance_ LTS** versions of Node.js]([nodejs.org/en/about/previous-releases#release-schedule](https://nodejs.org/en/about/previous-releases#release-schedule).
```
I think the link here makes sense!

Reviewer 2: There is an extra portion in the middle.

```suggestion
- Astro supports the [**latest _Maintenance_ LTS** versions of Node.js](https://nodejs.org/en/about/previous-releases#release-schedule).
```

> - Astro supports the [**latest _Maintenance_ LTS** versions of Node.js](https://nodejs.org/en/about/previous-releases#release-schedule).

Reviewer 4: Presumably this should also say that the current LTS is supported. Right now, this (and the line below) implies that Astro only supports Node 18, 19, and 21, but not 20.

Reviewer: Good call! I wasn't sure whether this line was meant to "assure about maintenance", but if it starts as "what we support", it should be exhaustive!

Author: Added in [`12dc2df` (#7652)](https://github.com/withastro/docs/pull/7652/commits/12dc2dfaf77278cfb837d0234ea338f7bb891cc9)

- Astro can support odd versions of Node.js.

#### Upgrade

The following rules define when Astro may deprecate, drop, or add support for versions of Node.js:

> - Odd versions of Node.js can be deprecated and/or dropped when the next even version of Node.js published. This change can occur in a **minor** release of Astro, after an arbitrary period of time decided by the Astro Core team.

Reviewer: ```suggestion
- Odd versions of Node.js can be deprecated and/or dropped when the next even version of Node.js published. This change can occur in a **minor** release of Astro, after a reasonable period of extended support as decided by the Astro Core team.
``` 
Maybe a bit friendlier?  :smile:

> - Upgrading the minimum **_extended_ LTS** version of Node.js can occur in a **minor** release of Astro.

Reviewer 2: By "Upgrading the minimum extended LTS version", does it mean an Astro minor can bump the minimum version only _within the same LTS_ (from the current `18.14.1` to `18.20.0` for example) or can it also bump to the next LTS once the current loses support (from `18` to `20` once support for Node 18 ends in June 2025)?

Author: Within the same LTS, as per your example

  - Security exception: If a security flaw in Node.js that **affects Astro** is disclosed and fixed, the Core team can bump the minimum version of the **_extended_ LTS** in a **patch** release.
- Upgrading minor or major versions of Node.js (**not** extended LTS) occurs only in major versions of Astro.
  - Security exception: If a security flaw in Node.js that **affects Astro** is disclosed and fixed, the Core team can bump the minimum version in a **minor** release.

### Extended maintenance
 

> The Core team will provide extended maintenance **for security fixes only** for one previous major version. This means that if the current major is `v4.*`, the Core team will back port security fixes and issue a new `v3.*` release. 

Reviewer: ```suggestion
The Core team will provide extended maintenance **for security fixes only** for one previous major version. This means that if the current major is `v4.*`, the Core team will back port security fixes and issue new `v3.*` releases when necessary. 
```

  Once these features are out of their experimental period, they will follow the normal semantic versioning contract.

- **Improvements to the documentation** (e.g. reference and error messages). They are built from source for the `docs` repository. This allows Astro to quickly update docs fixes and improvements in the cases where documentation source content is stored in the main `astro` repository.

### Node.js support and upgrade policies

#### Support

> - Astro supports the [**latest _Maintenance_ LTS** versions of Node.js](https://nodejs.org/en/about/previous-releases#release-schedule).

Author: ```suggestion
- Astro supports the [**latest _Maintenance_ LTS** version of Node.js](https://nodejs.org/en/about/previous-releases#release-schedule).
```

> - Astro supports the [**_Active_ LTS** versions of Node.js](https://nodejs.org/en/about/previous-releases#release-schedule)

Reviewer 2: ```suggestion
- Astro supports the [**_Active_ LTS** version of Node.js](https://nodejs.org/en/about/previous-releases#release-schedule)
```

There is only one Active LTS version at any given time.

Author: Can you change the suggestion and add `current _Active_ LTS` instead?

Reviewer 4: > Can you change the suggestion and add `current _Active_ LTS` too?

You can edit people's messages to add it
